src/providers/krb5/krb5_child.c

```diff
@@ -3154,6 +3158,49 @@ static int k5c_setup(struct krb5_req *kr, uint32_t offline)
     return kerr;
 }
 
+static krb5_error_code validate_keytab(struct krb5_req *kr)
+{
+    krb5_error_code kerr;
+    char krb5_conf_keytab[MAX_KEYTAB_NAME_LEN];
+    char *path_start = NULL;
+
+    if (kr->keytab == NULL && (
+        kr->pd->cmd == SSS_PAM_AUTHENTICATE ||
```

Comment:
I think the `kr->pd->cmd` checks can be dropped or is there a case where you think it might cause issues?

Reply:
I saw it being used in `unpack_buffer()` and on the sender side - `create_send_buffer()`. I thought it would be logical to behave the same way. For example `unpack_buffer()` after checking `kr->pd->cmd` may allow `kr->keytab` to be NULL. In this case I guess we should not validate it. So instead of having some flag checking operation mode I just follow the same rules with checking kr->pd->cmd.